### PR TITLE
Issue-5 - Add `Race` method

### PR DESCRIPTION
**Pull request by elycruz**
Tests and initial implementation.

Related issues: #2 #5 

### Tests cases - Usage:
  - With no promises.
  - With one promise, a passing one.
  - With multiple passing promises.
  - With one promise, a failing one.
  - With multiple promises and one failing.
  - With multiple failing promises.

### CODE REVIEW

**Comment by elycruz** [comment]
Code in this PR needs a refactor.  Will reopen when code is ready.

**Comment by elycruz** [comment]
PR is updated and ready for reopen.

**Comment by chebyrash** [comment]
The implementation is surprisingly similar to ```All```
The only difference is that we don't need a loop at the end, but simply return from any first case chosen by the ```select``` statement

**Comment by elycruz** [comment]
Yeah I saw this too, even the same looping logic for checking the promises (with difference of not keeping track of the index),

**Comment by chebyrash** [comment]
Yeah, exactly
Should be easy

**Comment by elycruz** [comment]
Yeah, about that last case (no need for last for loop in 'Race' method) hadn't even thought about (possibly for running on E (empty) lol).  Will update it this evening👍,
(cheers)

**Comment by chebyrash** [comment]
I think I managed to get the implementation done. We should have a look at the tests and move on to AllSettled

**Comment by elycruz** [comment]
@chebyrash K, gotcha, will take a look at your implementation (as I also had finished my implementation though I couldn't get travis-ci to stop saying "GOMODULE111=auto" (which was in turn causing errors with passing tests on there)).

**Comment by elycruz** [comment]
Yeah, the implementation is correct, adding my tests in (to edlc_dev)..,

**Comment by elycruz** [comment]
Aha!  Great!  Pull request passed!  @chebyrash

File: promise.go [hunk 1]
```diff
@@ -56,6 +58,14 @@ type Promise struct {
 	wg *sync.WaitGroup
 }
 
+// Result is used by methods that need to return result
+// 	objects (namely `AllSettled` method)
+type Result struct {
```
Comment (chebyrash) [inline promise.go]: Can we keep the unneeded commits out of this PR and focus only on the Race method. We push this into another PR for the AllSettled method

Comment (elycruz) [reply promise.go]: Sure, (:-), just saw this jehjeh, will make the change :+1: ),

**Pull request merged**